Cython/Compiler/ExprNodes.py

```diff
@@ -9442,15 +9442,19 @@ def generate_cyfunction_code(self, code):
         if self.defaults_tuple:
             code.putln('__Pyx_CyFunction_SetDefaultsTuple(%s, %s);' % (
                 self.result(), self.defaults_tuple.py_result()))
-        if self.defaults_kwdict:
-            code.putln('__Pyx_CyFunction_SetDefaultsKwDict(%s, %s);' % (
-                self.result(), self.defaults_kwdict.py_result()))
-        if def_node.defaults_getter:
-            code.putln('__Pyx_CyFunction_SetDefaultsGetter(%s, %s);' % (
-                self.result(), def_node.defaults_getter.entry.pyfunc_cname))
-        if self.annotations_dict:
-            code.putln('__Pyx_CyFunction_SetAnnotationsDict(%s, %s);' % (
-                self.result(), self.annotations_dict.py_result()))
+        if not self.specialized_cpdefs:
+            # disable introspection functions for fused dispatcher function since the user never sees it
```

Comment:
Is that actually true? The dispatch function should be what Cython inserts in place of the user provided function, so it seems to me that it's _more_ visible than the specialisations. Do we _have_ to disable them?

Reply:
Ah, hang on – the dispatch function is actually _not_ what users see, right? Instead, they do see a function with the expected signature(s). Yes, I think then it's ok to just leave these out.

Reply:
There's certainly tests for the introspection in `fused_def` so I'm relatively confident it didn't break anything major.